### PR TITLE
remove leading and trailing slash in auto-update basePath

### PR DESCRIPTION
**Pull request by pvnr0082t**
PR to fix basePath with leading and trailing slash in auto-update 
cc @PeterDaveHello 

Affected 434 libs as follow:

- 16pixels
- absurd
- accounting.js
- ajile
- alasql
- algoliasearch
- allow-me
- ally.js
- analytics.js
- angular-autofields
- angular-azure-mobile-service
- angular-busy
- angular-cache
- angular-css
- angular-data
- angular-dialog-service
- angular-filter
- angular-formly-templates-bootstrap
- angular-formly
- angular-gravatar
- angular-highlightjs
- angular-i18n
- angular-leaflet-directive
- angular-local-storage
- angular-material
- angular-material-icons
- angular-materialize
- angular-md5
- angular-moment
- angular-nvd3
- angular-recaptcha
- angular-retina
- angular-route-segment
- angular-schema-form
- angular-spinner
- angular-strap
- angular-ui-router
- angular-ui-tree
- angular-vertxbus
- angular-wizard
- angular-xeditable
- angulartics-google-analytics
- AniJS
- anima
- annyang
- api-check
- architect
- atmosphere
- audio5js
- autocomplete.js
- awesome-bootstrap-checkbox
- backbone-associations
- backbone-forms
- backbone-localstorage.js
- backbone-pageable
- backbone-react-component
- backbone-relational
- backbone.babysitter
- backbone.collectionView
- backbone.js
- backbone.layoutmanager
- backbone.validation
- bacon.js
- barn
- Base64
- basis.js
- basket.js
- bean
- benchmark
- bespoke.js
- bignumber.js
- bla
- blueimp-gallery
- blueimp-md5
- bootcards
- bootstrap-confirmation
- bootstrap-markdown
- bootstrap-select
- bootstrap-slider
- bootstrap-switch
- bootstrap-table
- bootstrap-tokenfield
- bootstrap-tour
- bricklayer
- bttrlazyloading
- caiuss
- camanjs
- cannon.js
- catiline
- chai
- chainloading
- chardin.js
- chroma-js
- cldrjs
- clipboard.js
- codemirror
- coffee-script
- Colors.js
- componentjs
- condition
- Cookies.js
- crel
- crossfilter
- cubism
- custom-elements-builder
- cutjs
- d3-geo-projection
- d3plus
- dancer.js
- danialfarid-angular-file-upload
- dat-gui
- dc
- depot
- device.js
- Director
- diva.js
- document-register-element
- dom4
- domainr-search-box
- domplotter
- draggabilly
- dropzone
- dustjs-helpers
- dustjs-linkedin
- eddy
- ekko-lightbox
- elasticsearch
- ember-dialog
- ember-i18n
- emblem
- es5-shim
- eve.js
- evil-icons
- fast-json-patch
- flickity
- floatthead
- fotorama
- free-jqgrid
- fullcalendar
- function-plot
- furtive
- fuse.js
- galleria
- geojs
- geojson2svg
- gh.js
- github-org-members.js
- gl-matrix
- golden-layout
- gorillascript
- grd
- gridly
- gsap
- Han
- handjs
- handlebars.js
- headhesive
- hiw-api
- horsey
- hprose-html5
- html5media
- hydra.js
- ICanHaz.js
- ice
- idbwrapper
- ie8
- iframe-resizer
- immstruct
- immutable
- ink
- insightjs
- intl-tel-input
- intro.js
- jcarousel
- jinplace
- jit
- jmpress
- jplayer
- jquery-bootgrid
- jquery-browser
- jquery-data-remote
- jquery-form-serializer
- jquery-image-upload
- jquery-impromptu
- jquery-jgrowl
- jquery-json-editor
- jquery-jsonview
- jquery-lazyload-any
- jquery-maskmoney
- jquery-migrate
- jquery-mobile
- jquery-noty
- jquery-once
- jquery-prompt21
- jquery-searcher
- jquery-sidebar
- jquery-smoove
- jquery-sortable
- jquery-ui-timepicker-addon
- jquery.age
- jquery.allowed-chars
- jquery.bootstrapvalidator
- jquery.devbridge-autocomplete
- jquery.fancytree
- jquery.formset
- jquery.gridster
- jquery.hashcash.io
- jquery.imagesloaded
- jquery.isotope
- jquery.lazyloadxt
- jquery.nanoscroller
- jquery.panzoom
- jquery.photocols
- jquery.postcodify
- jquery.simpleWeather
- jquery.smartmenus
- jquery.textcomplete
- jquery.tipsy
- jquery.ui-contextmenu
- jqueryui-touch-punch
- js-bson
- js-data-angular
- js-data-firebase
- js-data-http
- js-data-localforage
- js-data-localstorage
- js-sequence-diagrams
- jsface
- jsfeat
- jsforce
- jshint
- json3
- jsonld
- jsonlint
- jss
- jssip
- jssor-slider
- jstree
- jszip
- justifiedGallery
- jxon
- kartograph-js
- kineticjs
- klass
- knockback
- knockout-paging
- knockout-pre-rendered
- knockout-validation
- knockout
- kwargsjs
- Ladda
- layzr.js
- leaflet-dvf
- leaflet-hash
- Leaflet.awesome-markers
- leaflet.markercluster
- leaflet
- leapjs
- legojs
- less.js
- libil
- lightcase
- list.js
- list.pagination.js
- localforage
- localStorage
- loglevel
- lunr.js
- lz-string
- m8tro-bootstrap
- machina.js
- maquette
- markdown.js
- matreshka
- medium-editor-custom-html
- metisMenu
- mindb
- mini-meteor
- minicart
- minitranslate
- mithril
- mo
- mocha
- mojio-js
- moment-duration-format
- moment-range
- moment-timezone
- moment.js
- motajs
- msl-client-browser
- musicmetadata
- mvw-injection
- myscript
- nanobar
- nanogallery
- neo-async
- ng-clip
- ng-csv
- ng-dialog
- ng-flow
- ng-pdfviewer
- ng-showdown
- ng-slider
- ng-tags-input
- ngInfiniteScroll
- nomnoml
- notify
- nuclear-js
- numeral.js
- nwmatcher
- ol3
- omniscient
- onsen
- opentype.js
- operative
- orb
- ouibounce
- p2.js
- packery
- pangu
- parsley.js
- peerjs
- pickadate.js
- picturefill
- PikaChoose
- pixi.js
- placeholders
- plastiq
- platform
- playlyfe-js-sdk
- playlyfe-odysseus
- postal.js
- pouchdb
- primish
- probtn
- proj4js
- prostyle
- pubsub-js
- pusher-angular
- qoopido.js
- queue-async
- qwery
- ractive-require
- ractive
- ramda
- rangeslider.js
- react-bootstrap
- react-foundation-apps
- react-intl
- react-modal
- react-redux
- react-router-redux
- react-slick
- ready.js
- redux
- requirejs-plugins
- reqwest
- restangular
- rickshaw
- rivets
- rxjs
- script.js
- ScrollMagic
- seajs
- seedrandom
- select2-bootstrap-css
- set-iframe-height
- shaka-player
- shariff
- shred
- sidr
- signature_pad
- simple-statistics
- sizzle
- skrollr
- slick-carousel
- sockjs-client
- soundplayer-widget
- spa.js
- speakingurl
- spf
- sprintf
- sprite-js
- stacktrace.js
- stapes
- startbootstrap-sb-admin-2
- stomp.js
- strapdown-topbar
- string_score
- string.js
- SVG-Morpheus
- svg.draggy.js
- svg.pan-zoom.js
- swagger-ui
- sweetalert
- swig
- Swiper
- tabcomplete
- tabletop.js
- taffydb
- templatebinding
- textAngular
- then-request
- timecircles
- tinymce
- twitter-bootstrap
- typeahead.js
- typed.js
- typescript
- typicons
- UAParser.js
- underscore-contrib
- unsemantic
- upb
- urljs
- usertiming
- validatorjs
- vault.js
- vega
- vegas
- velocity
- videomail-client
- virtual-keyboard
- websqltracer
- whereyat
- wow
- xdomain
- xls
- xlsx
- yasgui
- yasqe
- yasr
- zeroclipboard
- zurb-ink

### CODE REVIEW

**Comment by PeterDaveHello** [comment]
This should come with the "test".

**Comment by pvnr0082t** [comment]
I will add test later.

**Comment by pvnr0082t** [comment]
@PeterDaveHello There are some libs don't have `basePath` in auto-update, so I add [Line276](https://github.com/pvnr0082t/cdnjs/blob/b0227b7e5f510e9a4c836f00f009946cd17ad22b/test/valid-packages-test.js#L276) and [Line287](https://github.com/pvnr0082t/cdnjs/blob/b0227b7e5f510e9a4c836f00f009946cd17ad22b/test/valid-packages-test.js#L287) to ignore the libs without `basePath`. I think we could open an issue. How do you think?

Here are the libs without basePath in auto-update:
- bootstrap-notify
- es6-shim
- FileSaver.js
- flexslider
- jquery-visibility
- mori
- portal
- processing.js
- punycode 
- q.js

**Comment by PeterDaveHello** [comment]
@pvnr0082t For the libs without basePath, may be mistakes, let's fix them and add test first!

**Comment by PeterDaveHello** [comment]
#9199 merged, let's get this merged ASAP! Please fix the conflicts first.

**Comment by pvnr0082t** [comment]
@PeterDaveHello @cdnjs/intern2 I have fixed the conflict. Please help me review it again, thank you.

**Comment by pvnr0082t** [comment]
@PeterDaveHello @cdnjs/intern2 I have rebased it on the latest master branch. Please help me review it again, thank you.

File: tools/fixFormat.js [hunk 1]
```diff
@@ -67,6 +67,21 @@ async.each(packages, function(item, callback) {
...
   }
+  if (pkg.npmFileMap) {
+    for (var i in pkg.npmFileMap) {
+      var npmbasePath = pkg.npmFileMap[i].basePath;
+      if (npmbasePath && npmbasePath.length != 0 && ((typeof npmbasePath) == "string")) {
+          npmbasePath = (npmbasePath).replace(/^\/|\/$/g , "");
+          pkg.npmFileMap[i].basePath = npmbasePath;
+      }
+    }
+  } else if (pkg.autoupdate) {
```
Comment (PeterDaveHello) [inline tools/fixFormat.js]: Let's just use `if`, not `else if`, you don't know which one the user want to preserve if there are multi configs.

File: tools/fixFormat.js [hunk 2]
```diff
@@ -67,6 +67,21 @@ async.each(packages, function(item, callback) {
     pkg.licenses = pkg.license;
     delete pkg.license;
   }
+  if (pkg.npmFileMap) {
+    for (var i in pkg.npmFileMap) {
+      var npmbasePath = pkg.npmFileMap[i].basePath;
+      if (npmbasePath && npmbasePath.length != 0 && ((typeof npmbasePath) == "string")) {
+          npmbasePath = (npmbasePath).replace(/^\/|\/$/g , "");
```
Comment (PeterDaveHello) [inline tools/fixFormat.js]: This pattern can not handle multiple slashes like `///dist///` which we don't want.

File: tools/fixFormat.js [hunk 3]
```diff
@@ -67,6 +67,21 @@ async.each(packages, function(item, callback) {
...
+          pkg.npmFileMap[i].basePath = npmbasePath;
+      }
+    }
+  } else if (pkg.autoupdate) {
+    var basePath = pkg.autoupdate.basePath;
+    if (basePath && basePath.length != 0 && ((typeof basePath) == "string")) {
+        basePath = (basePath).replace(/^\/|\/$/g , "");
```
Comment (PeterDaveHello) [inline tools/fixFormat.js]: Same problem here.

File: tools/fixFormat.js (hunk 2 above)
Comment (pvnr0082t) [inline tools/fixFormat.js]: @PeterDaveHello @cdnjs/intern2 I have fixed using `/^\/+|\/+$/g`. Please help me review it again, thank you.

File: tools/fixFormat.js (hunk 3 above)
Comment (pvnr0082t) [inline tools/fixFormat.js]: Same as npm part. I have fixed using `/^\/+|\/+$/g`. Please help me review it again, thank you.

**Comment by PeterDaveHello** [comment]
Nice work @pvnr0082t

**Pull request merged**